tests/holder/test_events.py

```diff
@@ -37,6 +38,18 @@ def mock_send_to_admins(mock_send_to_admins):
         yield mocked
 
 
+@pytest.fixture
+async def principal():
+    principal = Principal
```

Comment:
I think this should be `Principal()`, otherwise we're yielding the class Principal rather than an instance of it. We can also keep this simple and just directly yield the principal instance:
```python
@pytest.fixture
async def principal():
    yield Principal()
```